src/utils/tty.rs

```diff
@@ -1,42 +1,17 @@
-// Copied from rustc. isatty crate did not work as expected
-#[cfg(unix)]
-pub(crate) fn stderr_isatty() -> bool {
-    isatty(libc::STDERR_FILENO)
-}
+use once_cell::sync::OnceCell;
 
-#[cfg(windows)]
 pub(crate) fn stderr_isatty() -> bool {
-    isatty(winapi::um::winbase::STD_ERROR_HANDLE)
-}
-
-#[cfg(unix)]
-pub(crate) fn stdout_isatty() -> bool {
-    isatty(libc::STDOUT_FILENO)
+    // Cache the terminal detection for performance reasons.
+    // atty::is() is a relatively expensive IO operation.
```

Comment:
This should be cached in the Process abstraction, not here: because its process specific but we run multiple processes in parallel during testing.

It looks like the old code was likely overlooked / buggy.

Reply:
Fixed in https://github.com/rust-lang/rustup/pull/3351/commits/4b3d9e790ffe9770c2b03db0a561294e71cd29e2